vars/sonarExecuteScan.groovy

```diff
@@ -63,16 +63,21 @@ void call(Map parameters = [:]) {
                     if(!fileExists('.git')) utils.unstash('git')
                     piperExecuteBin.handleErrorDetails(STEP_NAME) {
                         writePipelineEnv(script: script, piperGoPath: piperGoPath)
-                        withSonarQubeEnv(stepConfig.instance) {
-                            withEnv(environment){
-                                influxWrapper(script){
-                                    piperExecuteBin.credentialWrapper(config, credentialInfo){
+                        withEnv(environment) {
+                            influxWrapper(script) {
+                                piperExecuteBin.credentialWrapper(config, credentialInfo) {
+                                    if (stepConfig.instance) {
+                                        withSonarQubeEnv(stepConfig.instance) {
+                                            echo "Instance is deprecated - please use serverUrl parameter to set URL to the Sonar backend."
+                                            sh "${piperGoPath} ${STEP_NAME}${customDefaultConfig}${customConfigArg}"
+                                            archiveArtifacts artifacts: "sonarscan.json", allowEmptyArchive: true
+                                            jenkinsUtils.handleStepResults(STEP_NAME, false, false)
+                                        }
+                                    } else {
                                         sh "${piperGoPath} ${STEP_NAME}${customDefaultConfig}${customConfigArg}"
-                                        archiveArtifacts artifacts: "sonarscan.json", allowEmptyArchive: true
                                     }
-                                    jenkinsUtils.handleStepResults(STEP_NAME, false, false)
-                                    readPipelineEnv(script: script, piperGoPath: piperGoPath)
                                 }
+                                readPipelineEnv(script: script, piperGoPath: piperGoPath)
```

Comment:
Now that we moved `withSonarQubeEnv` this could behave differently? In the past `readPipelineEnv` was inside `withSonarQubeEnv`. Is that a problem?

Reply:
This could be why we're seeing the warning, but I'm not sure. We want the readPipelineEnv for both cases (the old way and our new "instance" way), right?

Reply:
If I move this readPipelineEnv line back into the SonarQube wrapper (just as a test), the warning is gone!